### PR TITLE
fix #7909 feat(nimbus): optimize v6 api

### LINKED ISSUE #7909: V6 API Performance causing unnecessary server load
Looking at the server logs it looks like the V6 endpoint is taking ~10s per call and since jetstream calls it fairly regularly this can place an unnecessary amount of load on the database and servers.  We should get that down.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2902)
┆Link To Issue: https://mozilla-hub.atlassian.net/browse/EXP-2902

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* Server logs report the V6 API is running very slowly
* Looks like it's lagging on the changelog related call
* We recently added some model caches to solve the same thing for the V5 API
* But unlike V5, the V6 API also needs the enrollment end date and we didn't add a cache for that

This commit

* Adds a cache on the NimbusExperiment model for the enrollment end date
* Removes 'changes' from the prefetch call for the V6 API

### CODE REVIEW

**Comment by yashika-khurana** [comment]
the new circleci tests are not running 🤔

**Comment by jaredlockhart** [comment]
@yashika-khurana I updated the branch settings before they were merged.  All merged now, look at all the pretty tests! 😻

File: app/experimenter/docs/swagger-ui.html [hunk 1]
```diff
@@ -3291,6 +3291,9 @@
                 },
                 "schema": {
                   "type": "string"
+                },
+                "setsPrefs": {
+                  "type": "boolean"
```
Comment (yashika-khurana) [inline app/experimenter/docs/swagger-ui.html:3296]: `setsPrefs`?

**Review by yashika-khurana — CHANGES_REQUESTED** [review]
looks awesome, thank you @jaredlockhart . It completely makes sense to improve v6 api as we have seen the results in v5 🚀 , my only thing is that it includes one file change for `setsPrefs` which is not relevant to this PR, can you give a pass over that?

File: app/experimenter/docs/swagger-ui.html (hunk 1 above)
Comment (jaredlockhart) [reply app/experimenter/docs/swagger-ui.html:3296]: Yeah that's odd, I ran `make generate_docs generate_types` so it's weird that that was missed in @brennie 's PR, it should fail a check if there's pending updates to docs/types like that.  Let me try rerunning it.

Comment (jaredlockhart) [reply app/experimenter/docs/swagger-ui.html:3296]: Yeah see look, it's missed in main:

```
[app-9BVIRsKI-py3.9]jaredlockhart@Jareds-MacBook-Pro~/Documents/experimenter(main)$ git diff
diff --git a/app/experimenter/docs/openapi-schema.json b/app/experimenter/docs/openapi-schema.json
index 3615988e..598165f1 100644
--- a/app/experimenter/docs/openapi-schema.json
+++ b/app/experimenter/docs/openapi-schema.json
@@ -3649,4 +3649,4 @@
       }
     }
   }
-}
+}
\ No newline at end of file
diff --git a/app/experimenter/docs/swagger-ui.html b/app/experimenter/docs/swagger-ui.html
index 13a4cb59..2381fc83 100644
--- a/app/experimenter/docs/swagger-ui.html
+++ b/app/experimenter/docs/swagger-ui.html
@@ -3291,6 +3291,9 @@
                 },
                 "schema": {
                   "type": "string"
+                },
+                "setsPrefs": {
+                  "type": "boolean"
                 }
               },
               "required": [
@@ -3300,7 +3303,8 @@
                 "description",
                 "application",
                 "ownerEmail",
-                "schema"
+                "schema",
+                "setsPrefs"
               ]
             }
           },
[app-9BVIRsKI-py3.9]jaredlockhart@Jareds-MacBook-Pro~/Documents/experimenter(main)$ 
```

Comment (jaredlockhart) [reply app/experimenter/docs/swagger-ui.html:3296]: Filed https://github.com/mozilla/experimenter/issues/7911

Comment (yashika-khurana) [reply app/experimenter/docs/swagger-ui.html:3296]: thank you for your prompt response 🚀

**Review by yashika-khurana — APPROVED** [review]
Awesome, thanks @jaredlockhart 🙏 🎉

**Pull request merged**